### ISSUE TITLE
Utilities in tools/mac are a symbolic link outside the git repo

### ISSUE THREAD

**Issue by hzeller** [lucaseverini/ROPE#4]
The utilities in `tools/mac` can not be accessed, because they are actually symbolic links on your development machine :)

```
$ ls -l tools/mac/
total 8
lrwxrwxrwx 1 hzeller hzeller 67 Aug 31 22:06 autocoder -> /Users/Luca/Emulation/IBM1401/Autocoder/autocoder_f90_4.8/autocoder
lrwxrwxrwx 1 hzeller hzeller 77 Aug 31 22:06 ibm1401 -> /Users/Luca/Emulation/IBM1401/SimH/simh-master/Build/Products/Release/ibm1401
```

Speaking of which, where do I find the C/Fortan source for `autocoder` and `ibm1401` ?

**Comment by lucaseverini** [comment]
Let me look for them and I send you an email with the links to h.zeller@acm.org.
(Edit) I'm going to fix this issue putting the real files in place of the links.

**Comment by hzeller** [comment]
Still the source of the tools would be good as well. The autocoder space issue looked like something that could be fixed in the autocoder for instance. Having these tools on github would greatly help to improve them.
(I am considering volunteering at the Computer History Museum to help keep the 1401 ticking)

**Comment by lucaseverini** [comment]
I can send you the sources by email, I don't know if I can put them on github as I didn't write them.
I think I should ask them the permission to add the sources to this repo.
I will check for that. I don't know much about those issues.
However I can add the links to them in the description or somewhere else at least.

If you want to help the IBM1401 restoration team, I think you are more than welcome.
Robert Garner is the boss of the team.
You may find all the informations you need here. But you probably already know that.
http://ibm-1401.info/index.html

**Comment by hzeller** [comment]
Thanks, if you could send me the sources I'll have a first look.
I will then ask the authors if we can properly open-source it. Ideally it should be part of this repository, I agree.

**Comment by lucaseverini** [comment]
I placed a short text file explaining how to replace the broken links with the programs in tools/mac or by setting the preferences.

**Comment by lucaseverini** [comment]
The changes you made to build.xml don't let me buil/run/debug Rope from NetBeans IDE.
Can you fix that?

**Comment by hzeller** [comment]
mmh, maybe you need to roll it back then for now.

I changed the build.xml because it was not working at all with manual ant (it was including things that didn't exist and bailed out completely). For instance, it was including `nbproject/build-impl.xml` which is not in the project. So if you go back to the other build.xml, the other `build-impl.xml` should be included.

(I don't have NetBeans, I am the old-school kinda guy with just a command-line and an emacs :) ).

**Comment by lucaseverini** [comment]
Yes, unfortunately I'm a GUI guy. I can use command line but I don't love it.
I think the build.xml can be set in a way that lets both ant and netbeans to build/debug/run.

**Comment by lucaseverini** [comment]
I modified build.xml to work with ant and from netbeans.
I quickly changed your commands appending "_man" and everything now seems to work fine.
Let me know it it works for you too and feel free to fix it, keeping the netbeans stuff operational, if you want.

**Comment by hzeller** [comment]
ant still bails out, because it fails to include `nbproject/build-impl.xml` - is this just forgotten to be added to the repository ?

**Comment by lucaseverini** [comment]
I modified the .gitignore to let nbproject/build-impl.xml be added to the repository.
I test by cloning the repository anew and build using ant. Everything worked.
Please let me know it works for you too.
Feel free to rename the targets in build.xml if you think the postfix "_man" isn't so good.

**Comment by hzeller** [comment]
It looks much better now. It is not optimal yet, I will dive into that later.
One target that fails right now is `-post-jar` because it does not find the `tools/mac/autocoder` because that is a link into nirvana (the autocoder and ibm1401 are still the old links). The `how_to_replace_those_broken_links.txt` text references {REPO_ROOT}/dist/tools/mac, but that is not checked in (which is good, because it should be generated by the build file).

So if you have it in `dist/tools/mac` now, you need to move it in `tools/mac` and submit.

Once the build.xml is working fully with the original netbeans generated version, I am happy to get rid of the manual targets.

**Comment by lucaseverini** [comment]
Could you send me an ant script with all the possible build actions so I can use it to test if I'm doing it correctly?
Unfortunately I've never really used ant before.

**Comment by lucaseverini** [comment]
OK. Users first... ;-)
The symlinks where useful to me when I build new versions of SimH and Autocoder.
I can do things in a different way.
I put the real programs in mac/tools and removed the txt file.
Everything should be fine now.

Regarding the sources I send them to you in few minutes.

**Comment by hzeller** [comment]
Thanks, the build.xml now works!
I'll have a look at ROPE this weekend to see if there are some low-hanging fruits that I can help out from the ToDo list.

**Comment by lucaseverini** [comment]
Perfect!
I will review that list to remove typos, duplications and other mistakes as it wasn't meant to be used by other people than me.

**Comment by hzeller** [comment]
There seems to be one specific thing left specific to your environment: in `./nbproject/project.properties` it mentions `file.reference.rt.jar=/Users/Luca/Desktop/IBM1401/ROPE1401/rope1401/Libraries/rt.jar` This sounds like it is just the regular java rt.jar, so probably can be left out.

**Comment by lucaseverini** [comment]
Yes. It is.
On my computer rt.jar is in the library folder that is not in the repository and the path should be relative. 
I'm going to fix this asap.

Luca
